src/components/PartyDropdown.js

```diff
@@ -11,12 +11,24 @@ const searcher = new Fuse(parties, {
 })
 
 export default ({ partyId }) => {
-  const [state, setState] = useState({
-    dropdownOpen: !partyId ? true : false,
-    currentParty: partyId ? parties.find(p => p.id === partyId) : parties[0],
-  })
+  const [dropdownOpen, setDropdownOpen] = useState(!partyId ? true : false)
+  const [currentParty, setCurrentParty] = useState(
+    partyId ? parties.find(p => p.id === partyId) : parties[0]
+  )
+
   const [searchKeyword, setSearchKeyword] = useState("")
 
+  const dropdownRef = useRef()
+
+  const handleClickOutside = e => {
+    if (!!dropdownRef.current && !dropdownRef.current.contains(e.target)) {
+      setDropdownOpen(false)
+    }
+  }
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside)
```

Comment:
useEffect should return an unsubscribe function krub, otherwise there will be memory leak.

Reply:
fixed krub